examples/django/app/views.py

```diff
@@ -0,0 +1,39 @@
+import json
+from datetime import *
+import simplejson
+from django.http import HttpResponse, JsonResponse, HttpResponseNotFound
+from rest_framework import status
+from rest_framework.utils import json
+
+
+# Create your views here.
+city_list = ["pune", "austin", "santabarbara"]
+
+
+# API for Hello Application
+def hello(request):
+    return HttpResponse("Hello, Welcome to the Weather App!")
+
+
+# API for current server date
+def getdate(request):
+    current_datetime = datetime.now()
+    html = "Current Date and Time is: %s" % current_datetime
+    return HttpResponse(html, {"status": "success"}, status=status.HTTP_200_OK)
+
+
+# API for location details
+def get_location_details(request):
+    return HttpResponse(simplejson.dumps(city_list), content_type='application/json', status=status.HTTP_200_OK)
+
+
+# API for weather details for a location
+def get_weather_details(request, location_name):
+    file_name = location_name + ".json"
+    if location_name in city_list:
```

Comment:
What happens with `location_name` is not in `city_list`? What response will be sent? The code only returns a response if the `location_name` is supported. What happens with this curl:

```
curl -I "http://localhost:3000/weather/washington"
```

Reply:
The `location_name` is just a vari

Reply:
If `location_name` is in `city_list`, then there are two possible outcomes:
1. return HTTP 200 with JSON if we have a json file with the same name as the `location_name`
2. return a HTTP 404 ("Not Found") but with status of HTTP 500 - this is not correct to mix status codes

There is NO "else" condition to catch when `location_name` is NOT in `city_list`.

Reply:
@scottsbaldwin 

As mentioned in the previous two statements, the situation has improved.